### DISCUSSION TITLE
I use cmd script to process pictures but have different results? Why is this so?

### DISCUSSION THREAD

**Discussion by naohwen** [ImageMagick/ImageMagick#5318] [Digital Image Processing] [answered]
this is my cmd script；
`@SET a=%~nx1
gs -sDEVICE=jpeg -o temp.jpg -r200x200 %a%
pause
magick temp.jpg -negate -lat 20x20+5% -negate -colorspace Gray test2.jpg
pause`
If I use the script, the picture it outputs is black, but when I run the code line by line, the picture it outputs is white, why is this?
[test1.pdf](https://github.com/ImageMagick/ImageMagick/files/9091210/test1.pdf)
![test2](https://user-images.githubusercontent.com/82444429/178449674-23e38f52-d48c-4303-b602-ee45d72b59f5.jpg)

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform? Please always state this. I assume IM v7 on Windows.

That single line won't work. I suppose you have line-breaks somewhere, like this:
```
@SET a=%~nx1
gs -sDEVICE=jpeg -o temp.jpg -r200x200 %a%
pause
magick temp.jpg -negate -lat 20x20+5% -negate -colorspace Gray test2.jpg
pause
```
What do you mean by a "cmd script"? If you mean a BAT script, then the percent in the IM command needs to be doubled:
```
magick temp.jpg -negate -lat 20x20+5%% -negate -colorspace Gray test2.jpg
```

**Reply by naohwen** [reply]
Thank you so much.
